src/platform/nodejs/lib/util/http.ts

```diff
@@ -34,11 +34,10 @@ import { shallowEquals, throwMissingModuleError } from 'common/lib/util/utils';
 
 const globalAgentPool: Array<{ options: RestAgentOptions; agents: Agents }> = [];
 
-const handler = function (uri: string, params: unknown, client: BaseClient | null, callback?: RequestCallback) {
+const handler = function (uri: string, params: unknown, client: BaseClient | null) {
```

Comment:
Any reason not to move it to Http class as private method so it would have access to its `Http.client`?
`uri` and `params` are already not being used here, and if we were accessing client via `this.client`, we would be able to convert this higher-order function to a regular one and call it like `this._handler(null, res, res.body);`

Reply:
No reason not to do that, no. Doesn't seem necessary for this PR, though; I'd suggest raising a separate PR for it if it's a change you'd like to make.

Reply:
Opened PR https://github.com/ably/ably-js/pull/1627 for this